kernel/src/hil/nonvolatile_storage.rs

```diff
@@ -4,29 +4,29 @@ use crate::returncode::ReturnCode;
 
 /// Simple interface for reading and writing nonvolatile memory. It is expected
 /// that drivers for nonvolatile memory would implement this trait.
-pub trait NonvolatileStorage {
-    fn set_client(&self, client: &'static NonvolatileStorageClient);
+pub trait NonvolatileStorage<'client, 'buffer: 'client> {
```

Comment:
I don't there is a reason for these not to be the same lifetime. We should still be able to pass in a buffer with `&'static` lifetime, and it'll simply be upcast to whatever the `'client` liftetime is instantiated to